dotnet-logging/LoggingAPI/Tests/LoggingAPITests.cs

```diff
@@ -0,0 +1,32 @@
+using LoggingAPI;
+using Microsoft.Extensions.Logging;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using NSubstitute;
+using System;
+using System.IO;
+
+namespace Tests
+{
+    [TestClass]
+    public class LoggingAPITests
+    {
+
+        [TestMethod]
+        public void GivenLogger_ShouldLogInformation_WhenStudentIsCreated()
+        {
+            ILogger<Student> logger = Substitute.For<ILogger<Student>>();
+            Student student = new Student("Test", "Test department", logger);
+
+            logger.Received(1).LogInformation("A new student is created Test and his department is Test department");
+        }
+
+        [TestMethod]
+        public void GivenLoggerFactory_ShouldCreateLogger_WhenDepartmentIsCreated()
+        {
+            ILoggerFactory loggerFactory = Substitute.For<ILoggerFactory>();
+            Department department = new Department("Test", "Test department", loggerFactory);
+
+            loggerFactory.Received(1).CreateLogger<Department>();
```

Comment:
Use vars please.

Reply:
These tests seem a bit strange. You're using Received(1) as proof that it works? And shouldn't these two tests be the same. How do you actually know that the logger has been created in the second test?

Reply:
Updated to use var.

Received(1) works as proof because it will test whether a call to create a logger has been made or not. As to the second part of your question, since _logger is a private member, we cannot test whether it has been created or not. 

But as long as we are making a call to create a logger, we can be sure that the logger has been created. The purpose of this unit test is to test the department class not the logger api.

Reply:
Okay, thanks for clarifying that.